src/v2/api/index.md

```diff
@@ -1576,6 +1576,34 @@ type: api
   // `callback` is fired immediately with current value of `a`
   ```
 
+  Note that with `immediate` option you won't be able to unwatch the given property on the first callback call. 
+
+  ``` js
+  // This will cause an error
+  const unwatch = vm.$watch(
+    'value',
+    function() {
```

Comment:
Extreme nitpicking mode activated: I believe Vue CS leave a space between `function` and the opening bracket.